internal/controller/region/controller.go

```diff
@@ -273,9 +276,17 @@ func (r *Reconciler) removeHelmReleases(ctx context.Context, region *kcmv1.Regio
 	}
 
 	// If there are still HelmReleases left, requeue until cleanup is complete
-	if len(hrList.Items) > 0 {
+	slices.Sort(hrNames)
+	if len(hrNames) > 0 {
 		l.Info("Waiting for all HelmReleases to be deleted before removing finalizer")
-		return true, errs
+		meta.SetStatusCondition(&region.Status.Conditions, metav1.Condition{
+			Type:               kcmv1.ReadyCondition,
+			ObservedGeneration: region.Generation,
+			Status:             metav1.ConditionFalse,
+			Reason:             kcmv1.NotAllComponentsHealthyReason,
+			Message:            fmt.Sprintf("Waiting for all HelmReleases to be deleted: %s", hrNames),
+		})
+		return true, nil
```

Comment:
the condition will hang if it was set and then all of the releases are actually gone. is it okay, considering this step is the last before deletion from the cache and the server?

Reply:
It’s not critical since the Region object is removed almost immediately, but for consistency, I’ll clarify this in a future PR. Thanks.